compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/extended/RedundantVisibilityModifierChecker.kt

```diff
@@ -100,16 +106,22 @@ object RedundantVisibilityModifierChecker : FirBasicDeclarationChecker() {
         (symbol.fir as? FirMemberDeclaration)?.visibility?.let {
             return it
         }
+
         (symbol.fir as? FirPropertyAccessor)?.visibility?.let {
             return it
         }
+
         return null
     }
 
-    private val FirDeclaration.visibilityForDeclarationAndPropertyAccessor
+    private val FirMemberDeclaration.isLocalMember: Boolean
         get() = when (this) {
-            is FirMemberDeclaration -> status.visibility
-            is FirPropertyAccessor -> status.visibility
-            else -> null
+            is FirProperty -> this.isLocal
+            is FirRegularClass -> this.isLocal
+            is FirSimpleFunction -> this.isLocal
+            else -> false
         }
+
+    private val CheckerContext.lastInContainingDeclarationsVisibility
```

Comment:
Please rename it to `containingPropertyVisibility` or something like. Current name is very unclear.